### PR TITLE
First pass at a best practices document (fixes #188)

### LINKED ISSUE #188: What should a CDS Service do with this JWT?
There's some disparate guidance to CDS Services in the [Trusting EHRs](https://cds-hooks.org/specification/1.0/#trusting-ehrs) section of the spec now, about how the EHR-created JWT should be validated.

Currently, we say:
- CDS Services SHOULD whitelist the iss, jku and sub fields to only the EHRs they trust.

Additional best-practices to verify the JWT are:
1) Respect the exp field. A CDS Service SHOULD reject a request containing a JWT with an _exp_ value greater than now.
2) Use the jti / nonce field. A CDS Service SHOULD reject a request containing a _jti_ that has been previously used by the EHR.
3) Perhaps even reject requests with an old iat. A CDS Service MAY reject a request containing a _iat_ older than 8 minutes.
4) Verify that the public key used to sign the JWT is valid. A CDS Service SHOULD verify that the public key used to sign the JWT exists within the JWK set at the location identified by the jku and is identified by the kid. 

Is this worthwhile? What am I missing?

In a brief conversation with @kpshek , he pointed out that this type of guidance may be best placed in an implementation guide and not in the spec.

### PR DESCRIPTION
**Pull request by kpshek**
This is just a start but should be sufficient to resolve #188 I believe. More work needs to be done here for sure (even with additional security implementation best practices) but this provides a good starting point.

### CODE REVIEW

File: docs/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Implementation Best Practices
+
+This page serves as best practice guidance for CDS Hooks implementers. The best practices outlined here are not mandatory implementation rules; rather, they are suggested guidance. This is a living, in progress document and the best practices outlined here should not be considered absolute or complete.
+
+## Security
+
+Implementing the CDS Hooks security model requires perhaps the most thought and consideration for implementers. Security topics are never binary, are often complex, and robust implementations should factor in concepts such as risk and usability. Implementers should approach their security related development with thoughtful care.
```
Comment (isaacvetter) [inline docs/best-practices.md]: >Implementing the CDS Hooks security model requires perhaps the most thought and consideration for implementers.

What does this sentence mean? What is it trying to communicate? It should be removed or reworded to be more meaningful.

Comment (isaacvetter) [inline docs/best-practices.md]:
```suggestion
The CDS Hooks security model requires thought and consideration from implementers. Security topics are never binary, are often complex, and robust implementations should factor in concepts such as risk and usability. Implementers should approach their security related development with thoughtful care.
```

File: docs/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Verifying the JWT signature is a critical step in establishing trust of the caller of the CDS Service. As part of the whitelist of trusted CDS Clients, information on the public key(s) used by the CDS Client should also be stored. In some cases, this public key may be shared out-of-band. In other cases, the public key may be available at a remote endpoint and cycled on a regular basis. It is this latter case in which CDS Services should maintain their own rotating cache of public keys for the CDS Client.
+
+CDS Services should never store, share, or log JWTs to minimize the risk of theft and replay attacks. Information within the JWT (for instance, `iss`, `tenant`, `jti`) can be logged safely and is especially useful for analytics.
+
+If a CDS Service deems a JWT to be invalid for any reason, it should not leak the details of why the JWT failed validation back to the caller. If the caller were a malicious threat actor, leaking detailed information as to what was invalid may give the threat actor guidance on how to shape future attacks. Instead, responding to the request with a HTTP 401 Unauthorized response status code without any additional information is sufficient.
```
[the remark this answers is not in the document]
Comment (kpshek) [reply docs/best-practices.md]: Good point. Recommended in 5a84600

File: docs/best-practices.md (hunk 1 above)
Comment (kpshek) [reply docs/best-practices.md]: 👍

**Comment by brettmarquard** [comment]
Great spot to continuously improve!

**Pull request merged**